cassandra/io/twistedreactor.py

```diff
@@ -246,29 +212,48 @@ def __init__(self, *args, **kwargs):
         reactor.callFromThread(self.add_connection)
         self._loop.maybe_start()
 
-    def add_connection(self):
-        """
-        Convenience function to connect and store the resulting
-        connector.
-        """
+    def _check_pyopenssl(self):
         if self.ssl_options:
-
             if not _HAS_SSL:
                 raise ImportError(
-                    str(e) +
+                    str(import_exception) +
                     ', pyOpenSSL must be installed to enable SSL support with the Twisted event loop'
                 )
 
-            self.connector = reactor.connectSSL(
-                host=self.endpoint.address, port=self.port,
-                factory=TwistedConnectionClientFactory(self),
-                contextFactory=_SSLContextFactory(self.ssl_options, self._check_hostname, self.endpoint.address),
-                timeout=self.connect_timeout)
+    def add_connection(self):
+        """
+        Convenience function to connect and store the resulting
+        connector.
+        """
+        host, port = self.endpoint.resolve()
+        if self.ssl_context or self.ssl_options:
+            # Can't use optionsForClientTLS here because it *forces* hostname verification.
+            # Cool they enforce strong security, but we have to be able to turn it off
+            self._check_pyopenssl()
+
+            ssl_options = SSLCreator(
```

Comment:
nit: should this be named `ssl_connection_creator`

Reply:
Sure. It's a replacement for the `clientForTlsOptions` call which is why I named it that, but the parameter it takes in the `SSL4ClientEndpoint` call is called `sslContextFactory`. Twisted is weird. I'll just name it `ssl_connection_creator` :D

Reply:
damn twisted and their 234 compatible interfaces